src/editor/StyleEditor.cpp

```diff
@@ -33,6 +33,22 @@ struct InvalidColorError {
   QString setting;
 };
 
+template <typename T>
+void withSettingsValue(const QSettings &settings, const QString &key,
```

Comment:
Tiny thing, I don't think you need this withSettingsValue thing that takes in a variable function if you're only going to use it once. withSettingsColor was needed becuase in one case it was setting a QColor ref, and in the other it was setting a value in a QPalette

Reply:
may use it more than just for bool in the future. no harm in having a templated solution if it's the same amount of new code as writing a bool specific version

Reply:
oh wait do you mean the function parameter is not needed. i feel like we should probably keep it